sycl/plugins/unified_runtime/ur/adapters/cuda/enqueue.cpp

```diff
@@ -1230,24 +1257,27 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueUSMFill(
               UR_COMMAND_USM_FILL, hQueue, CuStream, StreamToken));
       EventPtr->start();
     }
+
+    auto N = size / patternSize;
     switch (patternSize) {
     case 1:
-      Result = UR_CHECK_ERROR(
-          cuMemsetD8Async((CUdeviceptr)ptr, *((const uint8_t *)pPattern) & 0xFF,
-                          size, CuStream));
+      Result = UR_CHECK_ERROR(cuMemsetD8Async(
```

Comment:
I know that this patch does not introduce it, but perhaps we could have a once over and get rid of all the: `Res = UR_CHECK_ERROR(...)` patterns.

Seems to me that `UR_CHECK_ERROR` is somehow awkward/misleading, as it can only ever return success (or throw/abort), it would be more clear if it was a void function/macro along the lines of `UR_SUCCESS_OR_DIE`.

The codebase is littered with if statements checking for the return status of `UR_CHECK_ERROR` that are never going to give back anything but success.

Reply:
good shout, tinkered with how that gets used here and above to account for it throwing instead of returning non-success codes